applications/API/serializers.py

```
@@ -0,0 +1,505 @@
+        return serializer_class(*args, **kwargs)
+
+    def bulk_serialize_response(self, queryset, extra_data: Optional[Dict] = None):
+        """
+        Serialize bulk data with additional metadata.
+        """
+        serializer = self.get_bulk_serializer(queryset, many=True)
+        response_data = {"results": serializer.data, "count": len(serializer.data)}
+
+        # Add extra metadata if provided
+        if extra_data:
+            response_data |= extra_data   
```

Pylint (3.3.7)

[convention] Trailing whitespace

(C0303)